dvc/api.py

```diff
@@ -91,6 +93,24 @@ def read(path, repo=None, rev=None, remote=None, mode="r", encoding=None):
         return fd.read()
 
 
+def get_params(stage: str, flatten: bool = False) -> Dict:
+    """Returns all the params associated with the given `stage`.
+
+    Args:
+        stage (str): Name of the dvc.yaml stage to get params from.
+        flatten (bool, optional): Whether to return flattened dict
+        or not. Defaults False.
+        If True, {"foo.bar": 1} would be returned, instead of
+        {"foo": {"bar": 1}}
+    """
+    repo = Repo()
+    params = {}
+    for dep in repo.index[stage].deps:
```

Comment:
Note that this may be too slow, you can use [`repo.stage.load_one(name=stage)`](https://github.com/iterative/dvc/blob/7708a8e292c2625d180b8b5569ef8f8be07e34ae/dvc/repo/stage.py#L286) which will only load that particular stage name from `dvc.yaml`.

There's also `repo.stage.get_target()` that is compatible with CLI.

If you are okay with getting multiple stages, there's also `repo.stage.load_all()` and `repo.stage.from_target()`.